app.js

```diff
@@ -394,6 +407,41 @@ const outputPortalReport = (req, res) => {
   });
 };
 
+const outputLogsCount = (req, res) => {
+  const json = validatePortalReportQueryJSON(req, res);
+  if (!json) {
+    // Error.
+    return;
+  }
+
+  let endpointValues = [];
```

Comment:
This chunk of code is duplicated in the `outputPortalReport` code - it might make sense to rename `validatePortalReportQueryJSON` to `getEndpoints` and return an object in the form of `{error, endpointValues, endpointMarkers}`.